internals/cli/cmd_run.go

```diff
@@ -239,9 +239,6 @@ out:
 		}
 	}
 
-	// Close our own self-connection, otherwise it prevents fast and clean termination.
-	rcmd.client.CloseIdleConnections()
```

Comment:
Please see the comment I've added at the client ```CloseIdleConnections()``` implementation. This call is typically needed in an iterative design where port / client allocations happen faster than the garbage collector free rate. For example, snapd uses this in some store tests, but in that context we should not use it using this API, just deal with it in the test code. I know this is an invasive change, but I would like to remove this, so that the Reponder interface does not need to provide this (unless of course this is needed).

Reply:
For reference, Juju uses `CloseIdleConnections` to ensure it closes the connection to Pebble immediately (I'm not entirely sure why! but it does). So yeah, I wouldn't get rid of this.

Reply:
@benhoyt I take your concern to heart, but I do feel like it should be possible to prove whether its needed or not in the Pebble case? The Juju case (I assume this is a separate code base, right) sounds exactly right, but the Pebble case makes this call just before the process terminates (in my analysis), which does not feel correct.

Reply:
You are right @benhoyt :) So gustavo recalled this, and the problem is solves, as stated in the comment (doh) is that it enables the server shutdown to go faster, as the server does not have to wait for idle connection timeout.